File: 2026_GFG_POTD/JANUARY/12-01-2026/K Sized Subarray Maximum.js
class Solution {
    maxOfSubarrays(arr, k) {
        const res = [];
        const dq = [];

        for (let i = 0; i < arr.length; i++) {
            while (dq.length && dq[0] <= i - k) dq.shift();
            while (dq.length && arr[dq[dq.length - 1]] <= arr[i]) dq.pop();
            dq.push(i);
            if (i >= k - 1) res.push(arr[dq[0]]);
        }

        return res;
    }
}

File: 2026_GFG_POTD/FEBRUARY/01-02-2026/K Sized Subarray Maximum.py
from collections import deque

class Solution:
    def maxOfSubarrays(self, arr, k):
        dq = deque()
        res = []
        
        for i in range(len(arr)):
            while dq and dq[0] <= i - k:
                dq.popleft()
            while dq and arr[dq[-1]] <= arr[i]:
                dq.pop()
            dq.append(i)
            if i >= k - 1:
                res.append(arr[dq[0]])
                
        return res